Always check that cgroup data is present

pugnascotia opened the pull request:
`OsProbe` fetches cgroup data from the filesystem, and has asserts that check for missing values. This PR changes most of these asserts into runtime checks, since at least one user has reported an NPE where a piece of cgroup data was missing.

Also update the testing guidelines to reflect current Gradle usage.

Closes #45396.

pugnascotia committed:
Always check that cgroup data is present

`OsProbe` fetches cgroup data from the filesystem, and has asserts that
check for missing values. This PR changes most of these asserts into
runtime checks, since at least one user has reported an NPE where
a piece of cgroup data was missing.

Also update the testing guidelines to reflect current Gradle usage.

Closes #45396.

elasticmachine commented:
Pinging @elastic/es-core-infra

pugnascotia committed:
Shorten long line, add JavaDoc.

A helper method in OsProbeTests breached the maximum line length. Break
up the line, and add some JavaDoc to the method.

jasontedor reviewed (dismissed):
I appreciate that you updated the testing instructions! However, the preference is to not mix concerns into a single commit. In general, it leads to larger commits, context switching during reviews, etc. And how can a passerby tracking changes they care about (e.g., docs, the build, or testing, etc.) know that *this* change contains changes to said docs? They'd have to read the diff and that defeats the purpose of summary lines in commit messages, PR titles, etc.

pugnascotia committed:
Revert testing doc changes for separate submission

pugnascotia commented:
No problem, I can split up the changes.

pugnascotia commented:
https://github.com/elastic/elasticsearch/pull/45617 for the docs changes.

jasontedor commented on server/src/main/java/org/elasticsearch/monitor/os/OsProbe.java:
```diff
@@ -511,17 +511,26 @@ boolean areCgroupStatsAvailable() {
                 final String cpuAcctControlGroup = controllerMap.get("cpuacct");
-                assert cpuAcctControlGroup != null;
+                if (cpuAcctControlGroup == null) {
+                    logger.debug("no [cpuacct] data found in control group stats");
```
I would suggest `cgroup` instead of `control group` since we refer to these as `cgroup` in JSON responses, etc.

jasontedor reviewed (commented):
I left one comment (that applies to three lines).

pugnascotia committed:
Use 'cgroup' abbreviation in log messages

When checking for particular items of cgroup data and finding them
missing, use the common abbreviaton 'cgroup' in the debug log messages.

jasontedor reviewed (approved):
LGTM.

colings86 commented:
It looks like the backporting is complete here so I have removed the backport pending label while generating 7.4.0 release notes. Please let me know if this is wrong
